### PR TITLE
add support for didDeployMessage (inspired by redis plugin)

### PR DESCRIPTION
**Pull request by sd**
The ember-cli-deploy-redis plugin provides a "didDeployMessage" config option that allows it to report the revisionKey that got deployed, but not activated, along with a simple explanation about the next step required for activation.

This PR follows the same pattern for s3-index, and defaults to reporting the following after deploy:
- Deployed but did not activate revision 268206c. To activate, run: ember deploy:activate production --revision=268206c

### CODE REVIEW

**Comment by lukemelia** [comment]
Thanks @sd. LGTM. It would be nice to have this (and the rest of this file) under test. WDYT @LevelbossMike?

**Comment by LevelbossMike** [comment]
LGTM as well. We should add a note about the new configuration setting in the README too.

I added some tests for index.js in https://github.com/ember-cli-deploy/ember-cli-deploy-s3-index/pull/15 We can add tests for the logging message there.

File: index.js [hunk 1]
```diff
@@ -26,6 +26,15 @@ module.exports = {
         s3Client: function(context) {
           return context.s3Client; // if you want to provide your own S3 client to be used instead of one from aws-sdk
         },
+        didDeployMessage: function(context){
+          var revisionKey = context.revisionData && context.revisionData.revisionKey;
+          var activatedRevisionKey = context.revisionData && context.revisionData.activatedRevisionKey;
```
Comment (LevelbossMike) [inline index.js:31]: Where is `revisionData.activatedRevisionKey` set? In the redis case this is set in the activate hook https://github.com/ember-cli-deploy/ember-cli-deploy-redis/blob/aaea5fb1ba806335e3dfbd56f383ae7728f84549/index.js#L89 which would be easy to do in this plugin as well but seems to be missing here

**Comment by sd** [comment]
See kids? this is what happens when you try to make a quick pull request, specially one without tests... you forget things, you embarrass yourself, and your mistakes live forever in the git log.

**Comment by ghedamat** [comment]
@sd maybe squash in one commit after addressing @LevelbossMike comment? :p

thanks in the meantime! :)

File: index.js [hunk 2]
```diff
@@ -70,9 +79,24 @@ module.exports = {
         this.log('preparing to activate `' + revisionKey + '`', { verbose: true });
 
         var s3 = new S3({ plugin: this });
-        return s3.activate(options);
+        return Promise.resolve(s3.activate(options))
```
Comment (LevelbossMike) [inline index.js]: `s3.activate` returns a promise already so no need to `Promise.resolve` here

File: index.js [hunk 3]
```diff
@@ -70,9 +79,24 @@ module.exports = {
...
+          .then(function(){
+            return {
+              revisionData: {
+                activatedRevisionKey: revisionKey
+              }
+            };
+          })
+          .catch(this._errorMessage.bind(this));
```
Comment (LevelbossMike) [inline index.js]: There should already be an error reported to the console without catching here. I think `ember-cli-deploy` already takes care of this. I may be mistaken though.

**Comment by sd** [comment]
Added info to README and squashed the commits.

Not sure why my rebase/squashing pulled in the "Fix quickstart instructions in README.md" commit too, I hope that's not a problem.

File: index.js [hunk 4]
```diff
@@ -90,6 +113,11 @@ module.exports = {
             context.revisions = revisions;
           });
       },
+
+      _errorMessage: function(error) {
```
Comment (LevelbossMike) [inline index.js:119]: We can safely remove that I guess as it isn't used anymore in the latest changes.

**Comment by LevelbossMike** [comment]
Thx! I added a last small comment. About the squashing of the fix quickstart. it would be great if you could only submit the necessary changes in one commit without the quickstart change (maybe just redo the changes and submit a new merge request as it might get annoying to undo the rebase). As this was merged via pull request and I'm not sure if it would work otherwise without changing the history.

This is a great addition! sorry that this  takes so long to merge.

**Comment by ghedamat** [comment]
@sd ping?

let us know if you need anything (or help with the rebase)

**Comment by ghedamat** [comment]
@sd sorry to bug you again, would love to get this in as we approach a new release cycle for the main ember-cli-deploy project

**Comment by sd** [comment]
I'll make some time to work on this. Sorry for the delay.

**Comment by ghedamat** [comment]
no problem @sd , thanks instead!

**Pull request closed without merging**